File: RAII/PYQT5/Excepciones/leerCSV.py
from Persona import Persona

nueva_persona = None
lista_personas = []

palabras = []
es_primero = True

try:
    with open("datosPersonas.csv","r") as f:
        for linea in f.readlines():
            if es_primero:
                es_primero = False
            else:
                palabras = linea.split(",")
                print(palabras[1])

                nueva_persona = Persona(int(palabras[0]), palabras[1], palabras[2], palabras[3])

                lista_personas.append(nueva_persona)

    print(f"El tamaño de la lista es {lista_personas.__len__()} Personas\n")
    #print(f"El tamaño de la lista es {len(lista_personas)}")
    for p in lista_personas:
        print(f">>> {p.nombre}")
except:
    print("Error en la lectura de archivo")

File: RAII/JAVAFX/excepciones/LeerCSV.java
import java.util.ArrayList;
import java.util.Scanner;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;

public class LeerCSV{
    public static void main(String[] args){
        Persona nueva;
        ArrayList<Persona> listaPersonas = new ArrayList<Persona>();

        String[] palabras;
        String linea;
        boolean esPrimero = true;
        try{
            File archivo = new File("datosPersonas.csv");
            Scanner lector =  new Scanner(archivo);
            
            while(lector.hasNext()){
                if(esPrimero){
                    linea = lector.nextLine();    
                    esPrimero = false;
                }
                else{
                    linea = lector.nextLine();
                    //System.out.println(linea);
                    palabras = linea.split(",");
                    //System.out.println("El nombre es: " + palabras[1]);

                    nueva = new Persona(Integer.parseInt( palabras[0]), palabras[1], palabras[2], palabras[3]);

                    listaPersonas.add(nueva);
                }
            }

            System.out.print("La lista tiene " + listaPersonas.size() + " Personas");

            for (Persona p : listaPersonas) {
                System.out.println(">>> " + p.nombre);
            }
            lector.close();

        }catch(Exception e){
            System.out.println("El archivo no existe");
            e.printStackTrace();
        }
    }
}